packages/config/src/index.ts

```diff
@@ -95,8 +95,13 @@ class TruffleConfig {
     const current = this.normalize(this);
     const normalized = this.normalize(obj);
 
-    const newConfig = assignIn(
+    //Order matters because an object can overshadow its prototype
+    const newConfig = Object.assign(
       Object.create(TruffleConfig.prototype),
+      //first grab the prototypes
+      Object.getPrototypeOf(current),
+      Object.getPrototypeOf(normalized),
+      //then get the instance values
```

Comment:
Welp, order matters here because an object could shadow its prototype.

``` javascript
const assignIn = require("lodash.assignin");
function Foo (name) { this.name = name; }
Foo.prototype.times = "proto";

// prototype is the truthiest
const assignIn_1 = (o, ...sources) => {
  return Object.assign(o, ...sources.flatMap(s => [s, Object.getPrototypeOf(s)]))
}

// ownProperty is the truthiest
const assignIn_2 = (o, ...sources) => {
  return Object.assign(o, ...sources.flatMap(s => [Object.getPrototypeOf(s), s]))
}


console.log("object before proto");
const foo = new Foo('me');
console.log(assignIn_1({a: 'angst'}, foo)); // { a: 'angst', name: 'me', times: 'proto' }
foo.times = "from the shadows";
console.log(assignIn_1({a: 'angst'}, foo)); // { a: 'angst', name: 'me', times: 'proto' }


console.log("proto before object");
const boo = new Foo('me');
console.log(assignIn_2({a: 'angst'}, boo)); // { a: 'angst', times: 'proto', name: 'me' }
boo.times = "from the shadows";
console.log(assignIn_2({a: 'angst'}, boo)); // { a: 'angst', times: 'from the shadows', name: 'me' }

console.log("lodash");
const loo = new Foo('me');
console.log(assignIn({a: 'angst'}, loo));   // { a: 'angst', name: 'me', times: 'proto' }
loo.times = "from the shadows";
console.log(assignIn_2({a: 'angst'}, loo)); // { a: 'angst', times: 'from the shadows', name: 'me' }

```